framework/src/modules/chain/logic/state_store/account_store.js

```diff
@@ -0,0 +1,136 @@
+const _ = require('lodash');
+
+const defaultAccount = {
+	publicKey: null,
+	secondPublicKey: null,
+	secondSignature: false,
+	u_secondSignature: false,
+	username: null,
+	u_username: null,
+	isDelegate: false,
+	u_isDelegate: false,
+	balance: '0',
+	u_balance: '0',
+	missedBlocks: 0,
+	producedBlocks: 0,
+	rank: null,
+	fees: '0',
+	rewards: '0',
+	vote: '0',
+	nameExist: false,
+	u_nameExist: false,
+	multiMin: 0,
+	u_multiMin: 0,
+	multiLifetime: 0,
+	u_multiLifetime: 0,
+};
+
+class AccountStore {
+	constructor(accountEntity, { mutate } = { mutate: true }) {
+		this.account = accountEntity;
+		this.data = [];
+		this.updatedKeys = {};
+		this.primaryKey = 'address';
+		this.name = 'Account';
+		this.mutate = mutate;
+		this.originalData = [];
+		this.originalUpdatedKeys = {};
+	}
+
+	async cache(filter, tx) {
+		const result = await this.account.get(filter, {}, tx);
+		this.data = _.uniqBy([...this.data, ...result], this.primaryKey);
+		return result;
+	}
+
+	createSnapshot() {
+		this.originalData = _.clone(this.data);
+		this.updatedKeys = _.clone(this.updatedKeys);
+	}
+
+	restoreSnapshot() {
+		this.data = this.originalData;
+		this.updatedKeys = {};
+	}
+
+	get(primaryValue) {
+		const element = this.data.find(
+			item => item[this.primaryKey] === primaryValue
+		);
+		if (!element) {
+			throw new Error(
+				`${this.name} with ${this.primaryKey} = ${primaryValue} does not exist`
+			);
+		}
+		return element;
+	}
+
+	getOrDefault(primaryValue) {
+		let element = this.data.find(
```

Comment:
prefer to use `const`

Reply:
We initalize `element` variable if it doesn't exist. That's why I am using const.

Reply:
maybe something like
```
const element = this.data.find(xxx);
if (element)
  return element;

const defaultElement = { ...defaultAccount, xxx};
this.data.push(defaultElement)
return defaultElement;
```